src/shapes/text.class.js

```diff
@@ -737,12 +737,16 @@
      */
     getCompleteStyleDeclaration: function(lineIndex, charIndex) {
       var style = this._getStyleDeclaration(lineIndex, charIndex) || { },
-          styleObject = { }, prop;
-      for (var i = 0; i < this._styleProperties.length; i++) {
-        prop = this._styleProperties[i];
-        styleObject[prop] = typeof style[prop] === 'undefined' ? this[prop] : style[prop];
+          decl = { }, slate = this._styleProperties;
+      for (var i = slate.length - 1; i >= 0; --i) {
+        var prop = slate[i];
+        if (style[prop] != null) {
```

Comment:
well, about that... when order doesn't matter, a reversed loop is usually *faster* in JS;
as for `null`, well - it's shorter for one - but it's also *equivalent* (i.e. `null == undefined`)